tests/app/platform_stats/test_rest.py

```diff
@@ -238,3 +239,23 @@ def test_volumes_by_service_report(
                            'service_id': str(fixture['service_with_sms_within_allowance'].id),
                            'service_name': fixture['service_with_sms_within_allowance'].name,
                            'sms_chargeable_units': 0, 'sms_notifications': 0}
+
+
+def test_sms_provider_daily_volumes_report(admin_request, sample_template):
+
+    create_ft_billing('2022-03-01', sample_template, provider='foo', rate=1.5, notifications_sent=1, billable_unit=2)
+    resp = admin_request.get(
+        'platform_stats.sms_provider_daily_volumes_report',
+        start_date='2022-03-01',
+        end_date='2022-03-01'
+    )
+
+    assert len(resp) == 1
+    assert resp[0] == {
+        'day': '2022-03-01',
+        'provider': 'foo',
+        'sms_totals': 1,
+        'sms_fragment_totals': 2,
+        'sms_chargeable_units': 2,
+        'sms_cost': 3,
```

Comment:
You could make this a decimal to test the casting implicitly.

Reply:
hmmm i'm not sure i follow - do you mean `assert isinstance(resp[0]['sms_cost'], float)` or similar? because the admin_request fixture takes json response and parses it - so that it was able to complete succesfully implies that we are casting all decimals to alternative json serialisable numerics

Reply:
I was just thinking: `'sms_cost': 3.797` (or whatever).

Very optional though: you're right this confirms it's implicitly working. Using a proper floating point number would just rule out things like the cast only working for integers, or the cast truncating to an integer.

Reply:
ah right. Good point, that's a low effort change to the input vals. i'll do that 👍